### ISSUE TITLE
Move Otterscan types to alloy

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#7372]
### Describe the feature

These types can be reused by reth+anvil or any other server 

https://github.com/paradigmxyz/reth/blob/33c466d88c4bee67f0b4eb15b2ca1b2fc421b10b/crates/rpc/rpc-types/src/otterscan.rs#L2-L2

and we should move them to

https://github.com/alloy-rs/alloy/blob/66fa1924a4f9165a0842838c5150260d0dd6de30/crates/rpc-trace-types/Cargo.toml#L2

**Comment by nkysg** [comment]
I can try it

**Comment by nkysg** [comment]
If this https://github.com/alloy-rs/alloy/pull/418 pr accepted, I will commit a pr for reth.

**Comment by mattsse** [comment]
it's possible that this is currently blocked by https://github.com/paradigmxyz/evm-inspectors/pull/78

**Comment by nkysg** [comment]
> it's possible that this is currently blocked by [paradigmxyz/evm-inspectors#78](https://github.com/paradigmxyz/evm-inspectors/pull/78)
I found that there is another issue which described in this https://github.com/paradigmxyz/reth/issues/7379
